circleci-orbs/preview.yml

```diff
@@ -0,0 +1,94 @@
+description: Easily register and deploy a component to a new environment on the Architect Cloud, then destroy the component and environment when ready
+
+display:
+  home_url: https://architect.io
+  source_url: https://github.com/architect-team/architect-cli/circleci-orbs
+
+jobs:
+  create:
+    parameters:
+      email:
+        type: string
+      password:
+        type: string
+      account:
+        type: string
+      environment:
+        type: string
+      platform:
+        type: string
+      component_name:
+        type: string
+      component_file:
+        type: string
+        default: ./architect.yml
+      deploy_flags:
+        type: string
+        default: ''
+    executor: preview_executor
+    steps:
+      - run: npm install -g @architect-io/cli@0.7.4
+      - run: architect login -e << parameters.email >> -p << parameters.password >>
+      - run: architect register << parameters.component_file >> -t << parameters.environment >>
+      - run: architect environment:create << parameters.environment >> -a << parameters.account >> --platform << parameters.platform >>
+      - run: architect deploy --auto_approve -a << parameters.account >> -e << parameters.environment >> << parameters.component_name >>:<< parameters.environment >> << parameters.deploy_flags >>
+  destroy:
+    parameters:
+      email:
+        type: string
+      password:
+        type: string
+      account:
+        type: string
+      environment:
+        type: string
+      component_name:
+        type: string
+    executor: preview_executor
+    steps:
+      - run: npm install -g @architect-io/cli@0.7.4
```

Comment:
@tjhiggins what're your thoughts on version pinning down to the patch level here? Seems a bit restrictive for pushing updates.

Reply:
We did the same thing github actions. I was torn then and now. The only comp I have is githubs own actions like `actions/cache@v2` which only major version pin. Our spec has become less volatile. Can you npm install a specific major version like `npm install -g @architect-io/cli@0.7` or even `npm install -g @architect-io/cli@0.*`?

Reply:
You can with NPM by default. See this doc:
https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept

I'm all for limiting the depth of versioning for the published actions/orbs, but since we're constantly changing the CLI with updates, I'd hate to have to keep going back to all our orbs/actions and updating version pins like this that could have been upgraded silently. What I suggested offline to @ryan-cahill was that we pin to minor versions of the CLI while in beta (e.g. @architect-io/cli@0.7) and eventually major versions which in theory are the only ones with breaking changes. 

_Here's an interesting NPM version calculator for confirmation of interpretation: https://semver.npmjs.com/_